packages/test/src/TestRunner.ts

```diff
@@ -1,19 +1,35 @@
-const { createInterfaceAdapter } = require("@truffle/interface-adapter");
-const web3Utils = require("web3-utils");
-const Config = require("@truffle/config");
-const Migrate = require("@truffle/migrate").default;
-const { Resolver } = require("@truffle/resolver");
-const expect = require("@truffle/expect");
-const util = require("util");
-const fs = require("fs");
-const path = require("path");
-const debug = require("debug")("lib:testing:testrunner");
-const Decoder = require("@truffle/decoder");
-const Codec = require("@truffle/codec");
-const OS = require("os");
-
-class TestRunner {
-  constructor(options = {}) {
+import { createInterfaceAdapter } from "@truffle/interface-adapter";
+import web3Utils from "web3-utils";
+import Config from "@truffle/config";
+import Migrate from "@truffle/migrate";
+import { Resolver } from "@truffle/resolver";
+import * as expect from "@truffle/expect";
+import util from "util";
+import fs from "fs";
+import path from "path";
+import debugModule from "debug";
+const debug = debugModule("lib:testing:testrunner");
+import * as Decoder from "@truffle/decoder";
+import type { LogDecoding } from "@truffle/codec";
+import * as Codec from "@truffle/codec";
+import OS from "os";
+import BN from "bn.js";
+
+export class TestRunner {
+  public config: Config;
+  public logger: any;
+  public provider: any;
+  public can_snapshot: boolean;
+  public first_snapshot: boolean;
+  public initial_snapshot: any;
```

Comment:
Can this be changed to camel case? 

- `can_snapshot` guards whether snapshot-revert can occur.
- `first_snapshot` starts out true and guards taking-a-snopshot in `initialize`, and is set to `false`.
- `initial_snapshot` implies multiple snapshots can occur, but the code seems to only store 1

The names don't quite reflect the usage. It's quite possible `can_snapshot` and `first_snapshot` reflect the same abstract state.

Reply:
Happy to change to camel case! I don't see how they might be the same. I believe `can_snapshot` refers to whether or not the chain you are using is "snapshottable". Like Ganache can "roll back to a snapshot". Goerli, on the other hand, will not roll back although you can use it to run tests if you like.

Reply:
Oh wait, did you mean `initial_snapshot` and `first_snapshot`?

Reply:
`can_snapshot` is initialized as `false` in the constructor, then set to `true` within  ` fn initialize()`. I don't think it considers the underlying chain's capabilities, unless :thinking: it's implied by `fn snapshot()` throwing... yuck! If that's the case, then `isSnapshotable` might be a better name and documenting the behavior.